engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -85,79 +91,119 @@ class EngineJobExecutionActor(jobKey: BackendJobDescriptorKey,
+            callCachingActorProps = props
+          case None =>
+            log.warning("{}: Turning off cache reading for this job as no cache copying actor is available for this backend.", jobTag)
+            effectiveCallCachingMode = effectiveCallCachingMode.withoutRead
+        }
+      }
       effectiveCallCachingMode match {
         case activity: CallCachingActivity if activity.readFromCache =>
           initializeJobHashing(jobDescriptor, activity)
-          goto(CheckingCallCache) using JobDescriptorData(jobDescriptor, bjeaProps)
+          goto(CheckingCallCache) using updatedData
         case activity: CallCachingActivity =>
           initializeJobHashing(jobDescriptor, activity)
-          runJob(jobDescriptor, bjeaProps)
-        case CallCachingOff => runJob(jobDescriptor, bjeaProps)
+          runJob(updatedData)
+        case CallCachingOff => runJob(updatedData)
       }
     case Event(response: BackendJobPreparationFailed, NoData) =>
       context.parent forward response
       context stop self
       stay()
   }
 
-  // When CheckingCallCache, the FSM always has EJEAJobDescriptorData
+  private def addHashesAndStay(data: NoResponseData, hashes: CallCacheHashes): State = {
+    val updatedData = data.copy(hashes = Option(Success(hashes)))
+    stay using updatedData
+  }
+
   private val callCachingReadResultMetadataKey = "Call caching read result"
   when(CheckingCallCache) {
-    case Event(HashError(t), JobDescriptorData(jobDescriptor, bjeaProps)) =>
+    case Event(HashError(t), data: NoResponseData) =>
       writeToMetadata(Map(callCachingReadResultMetadataKey -> s"Hashing Error: ${t.getMessage}"))
       recordHashError(t)
-      runJob(jobDescriptor, bjeaProps)
-    case Event(CacheMiss, JobDescriptorData(jobDescriptor, bjeaProps)) =>
+      runJob(data)
+    case Event(CacheMiss, data: NoResponseData) =>
       writeToMetadata(Map(callCachingReadResultMetadataKey -> "Cache Miss"))
-      log.info(s"Cache miss for job ${jobDescriptor.key.call.fullyQualifiedName}, index ${jobDescriptor.key.index}")
-      runJob(jobDescriptor, bjeaProps)
-    case Event(CacheHit(cacheResultId), JobDescriptorData(jobDescriptor, _)) =>
-      writeToMetadata(Map(callCachingReadResultMetadataKey -> s"Cache Hit (from result ID $cacheResultId)"))
-      log.info(s"Cache hit for job ${jobDescriptor.key.call.fullyQualifiedName}, index ${jobDescriptor.key.index}! Copying cache result $cacheResultId")
-      lookupCachedResult(jobDescriptor, cacheResultId)
+      log.info("Cache miss for job {}", jobTag)
+      runJob(data)
+    case Event(hit: CacheHit, data: NoResponseData) =>
+      writeToMetadata(Map(callCachingReadResultMetadataKey -> s"Cache Hit (from result ID ${hit.cacheResultId})"))
+      log.info("Cache hit for {}! Fetching cached result {}", jobTag, hit.cacheResultId)
+      fetchCachedResults(data, jobKey.call.task.outputs, hit)
+    case Event(hashes: CallCacheHashes, data: NoResponseData) =>
```

Comment:
I actually don't understand your first comment.  I added the `CallCacheHashes` as extra protection in case that message came in before `CacheHit` or `CacheMiss` but I think per our discussion last week it's actually unnecessary.